### DISCUSSION TITLE
Random Record Review

### DISCUSSION THREAD

**Discussion by cspayne** [crystalyragui/MARC2RDA#470] [General]
@GordonDunsire @CECSpecialistI @SitaKB @szapoun @lake44me @tmqdeborah @corialanus @junghaelee @JianPLee @AdamSchiff @pennylenger @hinxcode @dchen077 @tynancuw @sarahruska 

I ran 412 random records provided by Junghae through the transform. [The input file is here](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/marcDatasets/412randomrecords.xml) and [output in both RDF/XML and TTL is in this folder](https://github.com/uwlib-cams/MARC2RDA/tree/main/Working%20Documents/transformationCode/outputDataForReview/20241104). The RDF/XML is the direct output of the transform, while the TTL serialization also removes duplicate triples, making it a bit shorter (although it does mess with the order of the entities, so some navigation of the file is necessary to find the entities related to each other).

You may have to download the files to review them because they are so large. 

Please provide any feedback or questions in this discussion!

**Comment by cspayne** [comment]
One issue I ran into during the transform was with generating IRIs from subject headings. 

```
       <marc:datafield tag="651" ind1=" " ind2="7">
            <marc:subfield code="a">Honolulu &lt;Hawaii, 1993&gt;</marc:subfield>
            <marc:subfield code="2">swd</marc:subfield>
       </marc:datafield>
       <marc:datafield tag="650" ind1="0" ind2="7">
            <marc:subfield code="a">Frau &lt;Motiv&gt;</marc:subfield>
            <marc:subfield code="2">swd</marc:subfield>
       </marc:datafield>
       <marc:datafield tag="651" ind1=" " ind2="7">
            <marc:subfield code="a">Tokio &lt;1999&gt;</marc:subfield>
            <marc:subfield code="2">swd</marc:subfield>
       </marc:datafield>
```

It appears that subject headings in 651 from swd may contain < and > symbols, which are not valid in IRIs. This may not be the only time we run into this issue. What I have done currently is add these two characters to the stripAllPunctuation() function. 

However, it is worth considering the use of the built-in XSLT function encode-for-uri(). This function escapes all characters except upper- and lowercase letters A–Z, the digits 0–9, the hyphen-minus character, the underscore, the period, and the tilde. This does mean that characters with diacritics are escaped as well, but we could comfortably assume that the IRIs we are minting are all valid and usable if we went with this option.

**Reply by GordonDunsire** [reply]
@cspayne: I think it is ok to use the built-in XSLT function for field 651, but there is an issue if it is applied to timespan values in other fields because the hyphen is a primary distinguishing character. For example 'Report 1950-' (a series) and 'Report 1950' (the first issue in the series) should not be de-duplicated. This should be discussed with cataloguing experts.

**Reply by cspayne** [reply]
The built-in function will not remove the hyphen! It will also not remove characters that are not valid in URI's but provides a valid replacement, so it will not cause issues with accidental de-duplication.

**Reply by GordonDunsire** [reply]
@cspayne: then I guess it's ok to use the built-in function :-)
